Dockerfile

```diff
@@ -23,52 +18,47 @@ RUN apt-get update && apt-get install -y  \
     quantum-espresso      \
   && rm -rf /var/lib/apt/lists/*
 
-# needed for jupyterlab
+# Install what is needed for Jupyter Lab.
 RUN apt-get update && apt-get install -y \
      nodejs                \
      npm                   \
   && rm -rf /var/lib/apt/lists/*
 
-
-# Quantum-Espresso Pseudo Potentials
+# Quantum-Espresso Pseudo Potentials.
 WORKDIR /opt/pseudos
 RUN base_url=http://archive.materialscloud.org/file/2018.0001/v3;  \
 wget ${base_url}/SSSP_efficiency_pseudos.aiida;                    \
 wget ${base_url}/SSSP_precision_pseudos.aiida;                     \
 chown -R root:root /opt/pseudos/;                                  \
 chmod -R +r /opt/pseudos/
 
-# install packages that are not in the aiidalab meta package
-# 'fastentrypoints' is to fix problems with aiida-quantumespresso plugin installation
-RUN pip3 install --upgrade         \
-    'fastentrypoints'              \
-    'tornado==5.1.1'               \
+
+# Install Python packages needed for AiiDA lab.
+RUN pip install --upgrade         \
     'jupyterhub==0.9.4'            \
-    'notebook==5.7.4'              \
-    'nbserverproxy==0.8.8'         \
     'jupyterlab==0.35.4'           \
-    'appmode-aiidalab==0.5.0.1'
-
-# enable nbserverproxy extension
-RUN jupyter serverextension enable --sys-prefix --py nbserverproxy
-
-# This already enables jupyter notebook and server extensions
-RUN pip3 install aiidalab==v19.11.0a2
+    'nbserverproxy==0.8.8'
 
-# activate ipython kernel
-RUN python3 -m ipykernel install
+# AiiDA lab package goes last, because it overrides tornado.
+RUN pip install --upgrade 'aiidalab==v20.02.0a1'
 
-# TODO: delete, when https://github.com/aiidalab/aiidalab-widgets-base/issues/31 is fixed
-# the fileupload extension also needs to be "installed"
-RUN jupyter nbextension install --sys-prefix --py fileupload
+# Activate ipython kernel.
+# RUN python3 -m ipykernel install
 
 # Enable nbserverproxy extension.
 RUN jupyter serverextension enable --sys-prefix --py nbserverproxy
-# Enables better integration with jupyterhub.
+
+# Enables better integration with Jupyter Hub.
 # https://jupyterlab.readthedocs.io/en/stable/user/jupyterhub.html#further-integration
+# Quite a slow part.
```

Comment:
Perhaps write:  takes ~X minutes to install

Reply:
done